### PR TITLE
cupyx/scipy/signal: add place poles

### PR DESCRIPTION
**Pull request by ev-br**
cross-ref gh-7403

### CODE REVIEW

File: tests/cupyx_tests/scipy_tests/signal_tests/test_ltisys.py [hunk 1]
```diff
@@ -540,3 +541,318 @@ def test_second_order(self, xp, scp):
...
+             -491, -1543, -686]
+        ).reshape(6, 5)
+        P = xp.array([-25.-29.j, -25.+29.j, 31.-42.j,
+                     31.+42.j, 33.-41.j, 33.+41.j])
+
+        fsf = scp.signal.place_poles(A, B, P)
+        return fsf.computed_poles
+
+    @pytest.mark.xfail(reason="numerical stability: scipy QR vs numpy QR")
+    @testing.numpy_cupy_allclose(scipy_name='scp')
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/signal_tests/test_ltisys.py:643]: Does relaxing the tolerance (`rtol` and `atol` of `numpy_cupy_allclose`) remove the xfail?

**Comment by ev-br** [comment]
Updated with skips of tests which seem brittle on CI (and locally). Hopefully skipping based on the scipy version would be enough.

File: tests/cupyx_tests/scipy_tests/signal_tests/test_ltisys.py (hunk 1 above)
Comment (ev-br) [reply tests/cupyx_tests/scipy_tests/signal_tests/test_ltisys.py:649]: Not really. This test is flaky, and failure modes depend on the scipy version. Comments in the scipy code indicate things start differing between scipy.linalg.qr and numpy.qr (?), so this whole function is a bit brittle.
So the last commit skips the test depending on the scipy version, based on what the CI seems to suggest.

**Comment by ev-br** [comment]
Rebased on main and tightened up a test which was failing on CI for no good reason.

**Comment by takagi** [comment]
Would you try to make it contiguous then sort?
```
E   AssertionError: Only cupy raises error
E   
E   Traceback (most recent call last):
E     File "/root/.local/lib/python3.9/site-packages/cupy/testing/_loops.py", line 45, in _call_func
E       result = impl(*args, **kw)
E     File "/tmp/tmp.YnnJbxVJui/tests/cupyx_tests/scipy_tests/signal_tests/test_ltisys.py", line 585, in test_real_2
E       poles.sort()
E     File "cupy/_core/core.pyx", line 854, in cupy._core.core._ndarray_base.sort
E     File "cupy/_core/core.pyx", line 872, in cupy._core.core._ndarray_base.sort
E     File "cupy/_core/_routines_sorting.pyx", line 33, in cupy._core._routines_sorting._ndarray_sort
E   NotImplementedError: Sorting non-contiguous array is not supported.
```

**Comment by takagi** [comment]
Ah, seems that scipy returns results with different dtype on Windows. `cupy.linux.cuda122`'s failure is not releated, so it's okay.

**Comment by ev-br** [comment]
> seems that scipy returns results with different dtype on Windows.

Grrrr. Given that the test uses `np.real_if_close`,  a minimal guess is that on windows the fp noise exceeds the default tolerance of `100 * eps`.

It's too bad that the CI logs do not  show the arrays themselves or max difference. I wonder what's the easiest way to check these values (I can push a commit with a print of course but that would need somebody to trigger the CI run).

https://github.com/cupy/cupy/pull/7666/files#diff-e20ce11193646698fc4b7ae40e1bfd240f663a2051c4e0c5520b05d6e045a4d3R584

**Comment by ev-br** [comment]
We checked with @andfoy 's local windows setup (thank you Edgar!) and we cannot reproduce the dtype failure. On his local system with the matching scipy version (scipy 1.11.3), `test_real_2` both CuPy and SciPy only produces real poles:
https://gist.github.com/andfoy/5ff15c9df67c2f8d392fe03574443727.

So we cannot reproduce the CI failure and frankly I'm tempted to just xfail the test.

**Comment by takagi** [comment]
Please mark xfail the tests, thanks for your effort!

**Review by takagi — APPROVED** [review]
LGTM!

**Comment by takagi** [comment]
The CI failure is not related.

**Comment by takagi** [comment]
Thanks @ev-br!

**Pull request merged**